packages/runtime-common/realm.ts

```diff
@@ -488,34 +503,51 @@ export class Realm {
     path: LocalPath,
     contents: string,
     clientRequestId?: string | null,
-  ): Promise<WriteResult[]>;
-  async write(
+  ): Promise<FileWriteResult> {
+    let results = await this._batchWrite(
+      new Map([[path, contents]]),
+      clientRequestId,
+    );
+    return results[0];
+  }
+
+  async writeMany(
     files: Map<LocalPath, string>,
     clientRequestId?: string | null,
-  ): Promise<WriteResult[]>;
-  async write(
-    pathOrFiles: LocalPath | Map<LocalPath, string>,
-    contentsOrClientRequestId: string,
-    maybeClientRequestId?: string | null,
-  ): Promise<WriteResult[]> {
-    let files = new Map<LocalPath, string>();
-    let clientRequestId: string | undefined | null;
-    if (typeof pathOrFiles === 'string') {
-      files.set(pathOrFiles, contentsOrClientRequestId);
-      clientRequestId = maybeClientRequestId ?? null;
-    } else {
-      files = pathOrFiles;
-      clientRequestId = contentsOrClientRequestId ?? null;
-    }
+  ): Promise<FileWriteResult[]> {
+    return this._batchWrite(files, clientRequestId);
+  }
+
+  private async _batchWrite(
+    files: Map<LocalPath, string>,
+    clientRequestId?: string | null, //check if clientRequestId effects anything
+  ): Promise<FileWriteResult[]> {
     await this.indexing();
-    let results: WriteResult[] = [];
+    let results: FileWriteResult[] = [];
     let urls: URL[] = [];
     for (let [path, content] of files) {
       let url = this.paths.fileURL(path);
-      this.sendIndexInitiationEvent(url.href);
+      this.sendIndexInitiationEvent(url.href); //check if this does anything
       await this.trackOwnWrite(path);
-      let { lastModified } = await this.#adapter.write(path, content);
-      results.push({ path, lastModified });
+      try {
+        let doc = JSON.parse(content);
+        if (isCardResource(doc.data)) {
+          let serialized = await this.fileSerialization(
```

Comment:
I moved `fileSerialization` into an internal implementation. There are consequences of this which to me make more sense. But open to debate. Since with SRP principles, we might want write to just write but that is not what we are doing with the current implementation anyway